feat: add document metadata margin support for PyMuPDF extraction

priyankeshh opened the pull request:
## Description
Adds support for fetching document-specific margins from database metadata to use with PyMuPDF extraction, eliminating hardcoded margin values.

## Changes
- **New endpoint**: `/extract_with_document/{document_id}` accepts document ID and fetches margins from database
- **Database integration**: Imports extralit-server metadata classes to query document margins
- **Pixel-to-point conversion**: Converts margin metadata (pixels) to PDF points for PyMuPDF
- **Graceful fallbacks**: Uses ExtractionConfig defaults when no document margins available
- **Bug fix**: Fixed missing `file_bytes` variable in existing `/extract` endpoint
- **Gitignore**: Added `*.pdf` to prevent committing PDF files

## Implementation Details
- Fetches margins from `document.metadata_.analysis_metadata.layout_analysis.margin_analysis.estimated_margins`
- Requires all margin values (left_px, top_px, right_px, bottom_px) to be present
- Returns `None` when margins not found instead of hardcoded defaults
- Integrates with existing RQ job workflow via document ID parameter

priyankeshh committed:
feat: add document metadata margin support

- Add /extract_with_margins endpoint accepting document-specific margins
- Fix existing /extract endpoint bug (missing data variable)
- Add DocumentMargins pydantic model for validation
- Add PDF files to gitignore

priyankeshh committed:
fix: fetch margins from document metadata instead of hardcoding

- Replace hardcoded margins with database lookup
- Add document_id parameter to endpoint
- Import extralit-server metadata classes
- Fetch margins from document.metadata_.analysis_metadata.layout_analysis

priyankeshh committed:
fix: eliminate all hardcoded values and fix variable names

- Return None when no document margins found instead of hardcoded defaults
- Use ExtractionConfig defaults when no document margins available
- Add proper pixel-to-point conversion (0.75 factor)
- Require all margin values to be present (no partial data)
- Fix variable name from data to file_bytes to match function signature

JonnyTran reviewed (changes requested):
@priyankeshh I noticed you're using `src/app.py` that's using FastAPI, and I think you haven't pulled in the latest code changes in the base branch you're working on. It shouldn't have this code structure.

The code for the passing in the document margins should be called from the `pymupdf_to_markdown_job` in `extralit_ocr/jobs.py`

priyankeshh committed:
feat: implement document-specific margin support in RQ jobs and add initial __init__.py

priyankeshh committed:
resolve merge conflict: remove obsolete src/app.py file

JonnyTran commented on src/extralit_ocr/jobs.py:
```diff
@@ -0,0 +1,193 @@
+
+from rq import get_current_job
+
+from ..extract import extract_markdown_with_hierarchy, ExtractionConfig
```
Can you use the full import path (e.g. from `extralit_ocr`)?

JonnyTran commented on src/extralit_ocr/jobs.py:
```diff
@@ -0,0 +1,193 @@
+_LOGGER = logging.getLogger(__name__)
+
+# Try to import database components for margin fetching
+try:
```
`extralit-hf-space` always have `extralit_server` as the dependency, so this try-except block should removed.

Btw, for development, can you not use .venv and use a shared python environment for both repos?

JonnyTran replied:
Thanks. Btw do you know why you can't use 2 separate venv for each repo?

JonnyTran replied:
yes that's correct. If you have separate venv in the `extralit-hf-space/` path, when you install the python package, only the `extralit_ocr` package is installed in the venv, so it won't have the `extralit_server` package installed.

JonnyTran replied:
The most recommended approach is to activate only 1 python environment for the entire Extralit development across repos. I personally use `micromamba` since venv tend to only work with the local directory where the code lives.

JonnyTran commented on src/extralit_ocr/jobs.py:
```diff
@@ -0,0 +1,193 @@
+        raise RuntimeError(f"Failed to download file from S3: {e}")
+
+
+def pymupdf_to_markdown_job(
```
You also removed the `@jobs` decorator here. It's needed so the RQ worker in the `extralit-server` can call `extralit_ocr.jobs.pymupdf_to_markdown_job`

priyankeshh committed:
fix: improve PyMuPDF job implementation

- Use full import path for extralit_ocr.extract imports
- Remove unnecessary try-except block for extralit_server dependencies
- Add @job decorator for proper RQ worker integration
- Clean up import ordering and type annotations

JonnyTran committed:
Merge branch 'develop' into feat/document-metadata-margins

JonnyTran committed:
refactor: enhance margin retrieval in OCR job with async processing

- Updated _get_document_margins to use AsyncSessionLocal for non-blocking database access.
- Integrated DocumentProcessingMetadata for improved metadata handling.
- Converted pymupdf_to_markdown_job to async, ensuring seamless integration with the updated margin retrieval process.
- Cleaned up logging and error handling for better clarity during execution.

JonnyTran committed:
feat: implement document margin configuration in markdown extraction job

- Added _get_document_margins function to retrieve document-specific margins from metadata.
- Integrated margin configuration into pymupdf_to_markdown_job, allowing for dynamic margin adjustments during markdown extraction.
- Enhanced error handling and logging for improved debugging and clarity.

JonnyTran committed:
delete redundant file

JonnyTran committed:
refactor: update document margin handling in OCR job

- Changed parameter type of _get_document_margins from Document to DocumentProcessingMetadata for improved clarity.
- Simplified margin retrieval logic and enhanced logging for better debugging.
- Ensured consistent handling of document metadata during markdown extraction process.

JonnyTran committed:
refactoring

JonnyTran committed:
refactor
